src/ecc_compact/mod.rs

```diff
@@ -222,6 +226,23 @@ impl IntoBytes for PublicKey {
     }
 }
 
+impl PartialEq for PublicKey {
+    fn eq(&self, other: &Self) -> bool {
+        self.0 == other.0
+    }
+}
+
+impl Hash for PublicKey {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        let encoded = self
+            .0
+            .as_affine()
+            .to_compact_encoded_point()
+            .expect("compact point");
```

Comment:
Just an interesting point. We could actually use https://doc.rust-lang.org/std/hint/fn.unreachable_unchecked.html as part of 1.57 to get the compiler to remove guaranteed to be unreachable branches.